### PR TITLE
Accident report

### PR DESCRIPTION
**Pull request by memargonz**
Please review accident report interface.
The environment.ts need to have a valid APIs for Google and Openweather services:

export const environment = {
    production: false,
    WEATHER_API_KEY :"",
    GOOGLE_API_KEY:"",
    GOOGLE_NEARBY_Endpoint : "https://maps.googleapis.com/maps/api/place/nearbysearch/json?",
    WEATHER_Endpoint : "https://api.openweathermap.org/data/2.5/weather?units=metric",
    PLACES_Endpoint: "https://maps.googleapis.com/maps/api/place/details/json?fields=name%2Cformatted_phone_number%2Cicon"
  };
  
Thanks

### CODE REVIEW

File: crash/src/app/accident-report-container/intake/intake.component.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    console.log("Field validations failed")
+    return;
+  }
+  console.log(
+    `You entered :  
+    ${this.form.controls['Party'].value},
+    ${this.form.controls['WeatherConditions'].value},
+    ${this.form.controls['Location'].value}`
+  );
+  console.log(this.Image1)
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/intake/intake.component.ts]: Please remove console logs

File: crash/src/app/accident-report-container/map/map.component.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    .subscribe(
+    (res: any)=>{
+      console.log( res.results); 
+     
+      /* this.spawnPlaces(res.results)
+         console.log( res.results);  */
+       });  
+   
+  }  
+  /*  private spawnPlaces(landmarks:any[]){
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/map/map.component.ts]: Please remove any commented functions if they're not going to be used

File: crash/src/app/accident-report-container/upload-images/upload.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  </div>
+
+  
+</div>
+
+<div>
+  <img [src]="preview" class="preview" />
+</div>
+
+@if (message) {
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/upload-images/upload.component.html:15]: When using conditional ifs in angular, please use *ngIf.

ie: <div *ngIf="message">{{ message }}</div>

File: crash/src/app/crash.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+     let queryParams = new HttpParams();
+     queryParams = queryParams.append("place_id",`${placeID}`);  
+     queryParams = queryParams.append("key",environment.GOOGLE_API_KEY);
+     return this.http.get(environment.PLACES_Endpoint,{params: queryParams,responseType : 'json'});
+}
+
+getWeather(latitude: number, longitude: number, radius: number): any{
+ 
+   /* console.log(`${environment.WEATHER_Endpoint}
```
Comment (ianrbaguio) [inline crash/src/app/crash.service.ts]: Please remove any commented console.logs or functions

**Review by ianrbaguio — CHANGES_REQUESTED** [review]
Looks good Mario, just need a few changes

File: crash/src/app/accident-report-container/upload-images/upload.component.html (hunk 3 above)
Comment (ianrbaguio) [reply crash/src/app/accident-report-container/upload-images/upload.component.html:15]: It looks like this is a new way of doing *ngIf from Angular 16, please ignore this comment

**Review by srtviji — APPROVED** [review]
Approved

File: crash/src/app/crash.service.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+getNearbyPlaces(latitude: number, longitude: number)  {
+
+
+  let queryParams = new HttpParams();
+  queryParams = queryParams.append("location",`${latitude},${longitude}`);
+  queryParams = queryParams.append("keyword","building");
+  queryParams = queryParams.append("type","business");
+  queryParams = queryParams.append("rankby","distance");
+  queryParams = queryParams.append("key",environment.GOOGLE_API_KEY);
+ console.log(environment.GOOGLE_NEARBY_Endpoint)
```
Comment (ianrbaguio) [inline crash/src/app/crash.service.ts]: Remove console logs

File: crash/src/app/accident-report-container/map/map.component.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          (click)="collectInformation()" *ngIf="panelOpenState">
+          Get Address & Weather
+        </button>
+      </mat-panel-title>
+
+    </mat-expansion-panel-header>
+
+    <div id="map" style="width: 50%; height: 50%">
+
+      <!-- <google-map height="600px"
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/map/map.component.html]: Please remove this html element if it's not going to be use anymore

File: crash/src/app/accident-report-container/intake/intake.component.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+setImage3(img:any){
+  this.Image3=img;
+}
+
+
+
+onSubmit(): void{
+  this.submitted = true;
+  if (this.form.invalid) {
+    console.log("Field validations failed")
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/intake/intake.component.ts]: remove console log, having just the return should be okay. In the case of invalid form, we can either disable the submit button or show an error message in the front-end

File: crash/src/app/accident-report-container/map/map.component.ts [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+    loader
+      .importLibrary('maps')
+      /*  
+      dynamic loading of map does not work
+      .then(({Map}) => {
+          new google.maps.Map(document.getElementById("map") as HTMLElement,this.options);
+       })  */
+      .catch((e) => {
+        // do something
+        console.log("error caught");
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/map/map.component.ts]: remove console logs, in the case of an error we can use `throw new Error() instead`

File: crash/src/app/accident-report-container/map/map.component.ts [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+        console.log(e);
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/map/map.component.ts]: same comment as above

File: crash/src/app/accident-report-container/map/map.component.ts [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+    for (let i = 0; i < addresses.length; i++) {
+      places += addresses[i].place_id + "&"
+    }
+
+  }
+  searchNearLandmark(latlng: any) {
+    this.crashservice.getNearbyPlaces(latlng.lat, latlng.lng)
+      .subscribe(
+        (res: any) => {
+          console.log(res.results);
```
Comment (ianrbaguio) [inline crash/src/app/accident-report-container/map/map.component.ts]: I think we can keep this one for now, but please try to avoid using console logs

**Pull request merged**